### PR TITLE
refac(CI): add session scenario classes

### PR DESCRIPTION
**Pull request by rich-bugsnag**
## Goal

add session testing scenario classes

## Testing

All new tests cover previous csharp layer test scenarios

### CODE REVIEW

File: features/csharp/csharp_sessions.feature [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    And the error is valid for the error reporting API sent by the Unity notifier
+    And the error payload field "session.id" equals the stored value "session_id"
+    And the error payload field "session.startedAt" equals the stored value "session_startedAt"
+
+  Scenario: When a new session is started the error uses different session information
+    When I run the game in the "NewSession" state
+    And I wait to receive 2 sessions
+    And I wait to receive 2 errors
+    And I sort the errors by the payload field "events.0.exceptions.0.message"
+    And I sort the errors by the payload field "session.startedAt"
```
Comment (twometresteve) [inline features/csharp/csharp_sessions.feature:134]: I'm guessing you meant for one of these to say `sort the sessions`?

File: features/csharp/csharp_sessions.feature [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    # Error 1
+    And the error is valid for the error reporting API sent by the Unity notifier
+    And the event "session.events.handled" equals 1
+    And the error payload field "events.0.session.id" equals the stored value "session_id_1" ignoring case
+    And I discard the oldest error
+    # Error 2
+    And the error is valid for the error reporting API sent by the Unity notifier
+    And the event "session.events.handled" equals 1
+    And the error payload field "events.0.session.id" equals the stored value "session_id_2" ignoring case
+    And the error payload field "events.0.session.id" does not equal the stored value "session_id_1"
```
Comment (twometresteve) [inline features/csharp/csharp_sessions.feature:152]: I guess this step is belt and braces, but isn't needed as you've already verified that session_id_1 and session_id_2 are different.

File: features/csharp/csharp_sessions.feature [hunk 3]
```diff
@@ -0,0 +1,152 @@
+Feature: Session Tracking
+
+Scenario Outline: Automatically receiving a session
```
[the remark this answers is not in the document]
Comment (rich-bugsnag) [reply features/csharp/csharp_sessions.feature:3]: Done 👍

File: features/csharp/csharp_sessions.feature [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      | Handled Error 2              | 2       | 0         |
+      | Unhandled Error 1            | 2       | 1         |
+
+  Scenario Outline: No Auto session when not in enabled release stage
+    When I run the game in the "SessionNotInReleaseStage" state
+    And I wait for 5 seconds
+    Then I should receive no sessions
+
+  Scenario: When a session is stopped the error has no session information
+    When I run the game in the "StoppedSessionEvent" state
```
[the remark this answers is not in the document]
Comment (rich-bugsnag) [reply features/csharp/csharp_sessions.feature:104]: Done 👍

File: features/fixtures/maze_runner/Assets/Scripts/Scenarios/Csharp/Sessions/ResumedSession.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        base.PrepareConfig(apiKey, host);
+        Configuration.AutoTrackSessions = false;
+    }
+
+    public override void Run()
+    {
+        Bugsnag.StartSession();
+        Bugsnag.Notify(new System.Exception( "Error 1"));
+        Bugsnag.PauseSession();
+        Bugsnag.ResumeSession();
```
[the remark this answers is not in the document]
Comment (rich-bugsnag) [reply features/fixtures/maze_runner/Assets/Scripts/Scenarios/Csharp/Sessions/ResumedSession.cs:20]: done

File: features/csharp/csharp_sessions.feature (hunk 1 above)
Comment (rich-bugsnag) [reply features/csharp/csharp_sessions.feature:134]: i did mean that but the tests passed so looks like i didn't need it anyway. Removed 👍

File: features/csharp/csharp_sessions.feature (hunk 2 above)
Comment (rich-bugsnag) [reply features/csharp/csharp_sessions.feature:152]: Removed 👍

**Review by Cawllec — APPROVED** [review]

**Pull request merged**